pkg/action/executor/k8s_controller.go

```diff
@@ -154,7 +152,19 @@ func (c *parentController) update(updatedSpec *k8sControllerSpec) error {
 		if c.ormClient == nil {
 			return fmt.Errorf("failed to execute action with nil ORMClient")
 		}
-		err = c.ormClient.Update(origControllerObj, c.obj, ownerInfo)
+		resourcePaths, err := getResourcePath(kind, updatedSpec)
+		if err != nil {
+			return fmt.Errorf("failed to execute action, Unable to get resource paths: %v", err)
+		}
+		ownerResources, err := c.ormClient.GetOwnerResourcesForSource(c.obj, ownerInfo, resourcePaths)
+		if err != nil {
+			glog.Warning(err.Error())
+			return fmt.Errorf("failed to execute action, Unable to get owner resources: %v", err)
+		}
+		err = c.ormClient.UpdateOwners(c.obj, ownerInfo, ownerResources)
+		if err != nil {
+			return fmt.Errorf("failed to execute action due to failure in updating owners: %v", err)
```

Comment:
Nit: 

Just do

```go
return c.ormClient.UpdateOwners(c.obj, ownerInfo, ownerResources)
```

Reply:
I feel may be keep the error message related to action failure and append the specific error we get from UpdateOwners incase of failure ?

```
if err != nil {
			return fmt.Errorf("failed to execute action due to : %v", err)
```
The reason for this is, In updateOwners we don't specify anything related to action failure in error message, those are specific to the operation. I feel this will help user to navigate to the issue easily in case of failures in action